### ISSUE TITLE
Lacking functionality for vector indexing of element type StratReprPeriod

### ISSUE THREAD

**Issue by Zetison** [sintefore/TimeStruct.jl#18]
Type Vector{StratReprPeriod} can not currently be used as index for a RepresentativeProfile, even though StratReprPeriod can. This can easily be resolved by adding StratReprPeriod to the list of accepted types in the [Base.getindex](https://github.com/sintefore/TimeStruct.jl/blob/674c93e85ac96f875a63ec418c9c3e8d26d3e33a/src/profiles.jl#L186) function on line [189](https://github.com/sintefore/TimeStruct.jl/blob/674c93e85ac96f875a63ec418c9c3e8d26d3e33a/src/profiles.jl#L189) of [src/profile.jl](https://github.com/sintefore/TimeStruct.jl/blob/674c93e85ac96f875a63ec418c9c3e8d26d3e33a/src/profiles.jl):
```
function Base.getindex(
    TP::TS.TimeProfile,
    inds::Vector{T},
) where {T<:Union{TimePeriod,StratReprPeriod,TimeStructure}}
    return [TP[i] for i in inds]
end
```

**Comment by JulStraus** [comment]
The issue is not at this point, but at a different spot, as `StratReprPeriod <: TimeStructure`. Hence, the solution here is not the best proposed approach. We are investigating solutions.

**Comment by Zetison** [comment]
> StratReprPeriod

`StratReprPeriod <: TimeStructure` returns `false`, so maybe that is the problem?

**Comment by JulStraus** [comment]
The problem here is that `StratReprPeriod` is a parametric type:

```julia 
StratReprPeriod{T,OP<:TimeStructure{T}} <: AbstractRepresentativePeriod{T} <: TimeStructure{T}
```

Hence, neglecting the parametric type is resulting in the the reported `false`. This can be checked by creating a time structure, i.e.:

```julia
repr = RepresentativePeriods(
        2,
        20,
        [0.2, 0.8],
        [SimpleTimes(5, 1), SimpleTimes(5, 1)],
    )
ts = TwoLevel(100, [repr, repr, repr]; op_per_strat = 1)

sp1_rp_1 = collect(repr_periods(ts))[1]
typeof(sp1_rp_1) <: TimeStruct.TimeStructure
```

**Comment by trulsf** [comment]
Fixed as part of PR #19
